### DISCUSSION TITLE
Micro 820 Arrays and DINT

### DISCUSSION THREAD

**Discussion by lewblabencol** [ottowayi/pycomm3#212] [Q&A]
With LogixDriver I’ve been successful in .write with any BOOL or DINT but I’ve had a hard time writing to a member of a BOOL array or a bit of a DINT. After reading the comments here, the DINT issue seems good for Compact/ControlLogix but can I get help for the Micro 800?

**Comment by ottowayi** [comment]
Can you share the [logs](https://docs.pycomm3.dev/en/latest/getting_started.html#logging) for your attempts?  I'm not aware of an issue with the Micro800's, but I also don't have a physical device so they're not part of my normal testing.

**Comment by lewblabencol** [comment]
Code:

`import pycomm3
from pycomm3 import LogixDriver
from pycomm3.logger import configure_default_logger
configure_default_logger(filename='C:/Users/Work Station/Desktop/pycomm3_1.log')

with LogixDriver('192.168.50.26') as plc:
    plc.write('_IO_EM_DO_00',1)
    plc.write('Bool_Bool',1)
    plc.write('Bool_DINT',2)
    plc.write('Bool_DINT.0',1)
    plc.write('Bool_Array[0]',1)`

Log:
[pycomm3_1.log](https://github.com/ottowayi/pycomm3/files/8218990/pycomm3_1.log)

print of .read after that program:
_IO_EM_DO_00, True, BOOL, None
Bool_Bool, True, BOOL, None
Bool_DINT, 2, DINT, None
Bool_DINT.0, False, BOOL, None
Bool_Array[0], None, None, Tag doesn't exist - Bool_Array

On the Bool_DINT, I have also tried setting True but that didn't work either.

**Comment by lewblabencol** [comment]
plc.info output:

{'vendor': 'Rockwell Automation/Allen-Bradley', 'product_type': 'Programmable Logic Controller', 'product_code': 180, 'revision': {'major': 12, 'minor': 11}, 'status': b'4\x00', 'serial': '60fa8f5e', 'product_name': '2080-LC20-20QWB', 'keyswitch': 'UNKNOWN', 'programs': {}, 'tasks': {}, 'modules': {}}
